markdownhighlighter.h

```diff
@@ -78,6 +82,8 @@ Q_OBJECT
         CodeNumLiteral = 1005,
         CodeBuiltIn = 1006,
 
+        CodeBlockBackgroundColor = 1007,
```

Comment:
I wonder why we need the CodeBlockBackgroundColor, don't we have `MarkdownHighlighter::CodeBlock` for this?

Reply:
I'll migrate to `CodeBlock`.

Reply:
That's because I was thinking of adding a separate setting for it in the Font color dialog.

Reply:
Do we need a 2nd background color for code blocks? 😁 
I now did this: https://github.com/pbek/qmarkdowntextedit/commit/9db742fdc51c6152115942c17a9039da4f896890
It also checks if the color was set at all (and fixes a typo)...

Reply:
It needs alpha value to provide a good look(not necessarily) in my opinion. Using code block color limits that, and if you turn bg color on in `CodeBlock`,the results look very weird (unless you turn it off so that the background color is only used for painting the rectangle)

Reply:
Can you please test the current commit https://github.com/pbek/QOwnNotes/commit/0eba10a75bdd941510624b18bbbd10c52d4041ae and elaborate on what you mean by "weird". Looking great in my opinion without alpha (just with a solid background color like we had for the "Light" schema). 😁

Reply:
Just tried, looking pretty good.

Reply:
Ok, great. I almost thought I overlooked something. 😬

Reply:
We'll need to tweak the background color of White color scheme. It doesn't fit very well I think :thinking:

Reply:
You think so? What is off?

Reply:
it's a little bluish on my side, not as visible as the gray background in dark mode.

Reply:
For me it's looking like before, but with the block using the full width.

![Screenshot 2019-12-27 at 07 26 58](https://user-images.githubusercontent.com/1798101/71504623-79095f80-287a-11ea-929e-4e1c55767161.png)

Reply:
ah yes, but the bluish  color is not as visible I think. But i guess its also personal taste. :laughing:

Reply:
Do you mean the light green background? What do you mean by not visible? I guess it shouldn't be very visible as background, maybe it's even too visible (not very high contrast between front and back) 😁

Reply:
i thought it was bluish. :rofl: 

I guess i am looking for something more akin to this:

![](https://user-images.githubusercontent.com/180129/70249879-0f020c80-174b-11ea-9e22-de539ed2c3fd.png)

or what github does

```cpp
int x = 10;
```

Reply:
I find the green... soothing. 😁